packages/fields/src/fields/template/template-edit.tsx

```diff
@@ -0,0 +1,245 @@
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useMemo, useState } from '@wordpress/element';
+// @ts-ignore
+import { parse } from '@wordpress/blocks';
+import type { WpTemplate } from '@wordpress/core-data';
+import { store as coreStore } from '@wordpress/core-data';
+import type { DataFormControlProps } from '@wordpress/dataviews';
+
+/**
+ * Internal dependencies
+ */
+// @ts-expect-error block-editor is not typed correctly.
+import { __experimentalBlockPatternsList as BlockPatternsList } from '@wordpress/block-editor';
+import {
+	Button,
+	Dropdown,
+	MenuGroup,
+	MenuItem,
+	Modal,
+} from '@wordpress/components';
+import { useAsyncList } from '@wordpress/compose';
+import { useSelect } from '@wordpress/data';
+import { decodeEntities } from '@wordpress/html-entities';
+import { __ } from '@wordpress/i18n';
+import { getItemTitle } from '../../actions/utils';
+import type { BasePost } from '../../types';
+
+function useTemplates( postType: string ) {
+	return useSelect(
+		( select ) =>
+			select( coreStore ).getEntityRecords< WpTemplate >(
+				'postType',
+				'wp_template',
+				{
+					per_page: -1,
+					post_type: postType,
+				}
+			) ?? [],
+		[ postType ]
+	);
+}
+
+export function useAllowSwitchingTemplates( {
```

Comment:
Same here, is this hook really needed?

Reply:
Fixed with 7cf284378c54a2fdbf0c43b9a68ff085bc0b7b21

Reply:
Did you actually remove this hook because I still see it in the diff :)

Reply:
I forgot to remove it - sorry 😅. Fixed with f891475bd3f69bcb66b1fdadff1e7173fdf5d424.